PowerShellTools/Classification/PowerShellTokenizationService.cs

```diff
@@ -51,136 +51,145 @@ public PowerShellTokenizationService(ITextBuffer buffer)
-	        var array = _generatedTokens;
+            ParseError[] errors;
+            _generatedAst = Parser.ParseInput(spanText, out _generatedTokens, out errors);
+
+            var position = spanToTokenize.GetStartPoint(Buffer.CurrentSnapshot).Position;
+            var array = _generatedTokens;
 
             Log.Debug("Classifying tokens.");
             _tokenSpans = _classifierService.ClassifyTokens(array, position);
 
             Log.Debug("Tagging error spans.");
-	        _errorTags = _errorTagService.TagErrorSpans(Buffer, position, errors).ToList();
+            // Trigger the out-proc error parsing only when it's enabled and there are errors from the in-proc parser
+            if (errors.Length !=0 && PowerShellToolsPackage.UseOutProc)
+            {
+                var errorsParsedFromOutProc = PowerShellToolsPackage.IntelliSenseService.GetParseErrors(spanText);
+                _errorTags = _errorTagService.TagErrorSpans(Buffer, position, errorsParsedFromOutProc).ToList();
+            }
+            else
+            {
+                _errorTags = _errorTagService.TagErrorSpans(Buffer, position, errors).ToList();
+            }
 
             Log.Debug("Matching braces and regions.");
             _regionAndBraceMatchingService.GetRegionsAndBraceMatchingInformation(spanText, position, _generatedTokens, out _startBraces, out _endBraces, out _regions);
-	    }
+        }
 
-	    private void SetTokenizationProperties()
-	    {
-	        SetBufferProperty(BufferProperties.Tokens, _generatedTokens);
+        private void SetTokenizationProperties()
+        {
+            SetBufferProperty(BufferProperties.Tokens, _generatedTokens);
             SetBufferProperty(BufferProperties.Ast, _generatedAst);
-	        SetBufferProperty(BufferProperties.SpanTokenized, null);
-	        SetBufferProperty(BufferProperties.TokenErrorTags, _errorTags);
-	        SetBufferProperty(BufferProperties.EndBrace, _endBraces);
-	        SetBufferProperty(BufferProperties.StartBrace, _startBraces);
-	        SetBufferProperty(BufferProperties.Regions, _regions);
-	        SetBufferProperty(BufferProperties.TokenSpans, _tokenSpans);
-	    }
-
-	    private void RemoveCachedTokenizationProperties()
-	    {
-	        Buffer.Properties.RemoveProperty(BufferProperties.RegionTags);
-	    }
-	}
+            SetBufferProperty(BufferProperties.SpanTokenized, null);
+            SetBufferProperty(BufferProperties.TokenErrorTags, _errorTags);
+            SetBufferProperty(BufferProperties.EndBrace, _endBraces);
+            SetBufferProperty(BufferProperties.StartBrace, _startBraces);
+            SetBufferProperty(BufferProperties.Regions, _regions);
+            SetBufferProperty(BufferProperties.TokenSpans, _tokenSpans);
+        }
+
+        private void RemoveCachedTokenizationProperties()
+        {
+            Buffer.Properties.RemoveProperty(BufferProperties.RegionTags);
+        }
+    }

```

Comment:
You can! :D
https://github.com/Microsoft/poshtools/pull/52/files?w=1

https://chrome.google.com/webstore/detail/github-diff-whitespace/lhbcdehjihmbiafeodkfnbndleijnnhp